src/facilities/costCenters.spec.js

```diff
@@ -0,0 +1,540 @@
+import omit from 'lodash.omit';
+import CostCenters from './costCenters';
+import * as facilitiesUtils from '../utils/facilities';
+
+describe('Facilities/CostCenters', function() {
+  let baseRequest;
+  let baseSdk;
+  let expectedHost;
+
+  beforeEach(function() {
+    this.sandbox = sandbox.create();
+
+    baseRequest = {
+      delete: this.sandbox.stub().resolves(),
+      get: this.sandbox.stub().resolves(),
+      post: this.sandbox.stub().resolves(),
+      put: this.sandbox.stub().resolves()
+    };
+    baseSdk = {
+      config: {
+        audiences: {
+          facilities: fixture.build('audience')
+        }
+      }
+    };
+    expectedHost = faker.internet.url();
+  });
+
+  afterEach(function() {
+    this.sandbox.restore();
+  });
+
+  describe('constructor', function() {
+    let costCenters;
+
+    beforeEach(function() {
+      costCenters = new CostCenters(baseSdk, baseRequest, expectedHost);
+    });
+
+    it('sets a base url for the class instance', function() {
+      expect(costCenters._baseUrl).to.equal(expectedHost);
+    });
+
+    it('appends the supplied request module to the class instance', function() {
+      expect(costCenters._request).to.equal(baseRequest);
+    });
+
+    it('appends the supplied sdk to the class instance', function() {
+      expect(costCenters._sdk).to.equal(baseSdk);
+    });
+  });
+
+  describe('addFacility', function() {
+    context('when all required information is supplied', function() {
+      let expectedFacilityId;
+      let expectedCostCenterFacility;
+      let expectedCostCenterId;
+      let formatCostCenterFacilityFromServer;
+      let promise;
+      let rawCostCenterFacility;
+      let request;
+
+      beforeEach(function() {
+        expectedCostCenterFacility = fixture.build('costCenterFacility');
+        expectedFacilityId = expectedCostCenterFacility.facilityId;
+        expectedCostCenterId = expectedCostCenterFacility.costCenterId;
+        rawCostCenterFacility = fixture.build('costCenterFacility', {
```

Comment:
If you did this as `rawCostCenterFacility = fixture.build('costCenterFacility', expectedCostCenterFacility, { fromServer: true });`, it should be the fixture to have the raw object to have the same values as the expected object (just with the different keys).

(This doesn't need to happen, just kind of an FYI and cool thing Rosie can do)

Reply:
BUT, this _*DOES*_ need `null` passed in though for it to generate the right format